oak_functions/abi/src/lib.rs

```diff
@@ -27,33 +27,110 @@ use core::mem::size_of;
 use serde::{Deserialize, Serialize};
 
 pub mod proto {
-    use alloc::vec::Vec;
     include!(concat!(env!("OUT_DIR"), "/oak.functions.abi.rs"));
     include!(concat!(env!("OUT_DIR"), "/oak.functions.lookup_data.rs"));
     include!(concat!(env!("OUT_DIR"), "/oak.functions.invocation.rs"));
+}
 
-    impl Response {
-        /// Creates a new instance of Response.
-        ///
-        /// Sets the `status` and `body` to the given status and body, and sets the
-        /// `length` to the length of the body.
-        pub fn create(status: StatusCode, body: Vec<u8>) -> Self {
-            Response {
-                status: status as i32,
-                body: body.clone(),
-                length: body.len() as u64,
-            }
-        }
+/// See REQUEST_RESPONSE_ENCODING.MD in the crate root.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Request {
+    pub body: alloc::vec::Vec<u8>,
+}
+
+/// See REQUEST_RESPONSE_ENCODING.MD in the crate root.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Response {
+    pub status: StatusCode,
+    pub body: alloc::vec::Vec<u8>,
+    pub length: u64,
```

Comment:
Do we need this length field at all? In `Response::create`, below, you set it to `body.len()`, and I don't think it's ever set to anything other than that.

Reply:
Added a comment to clarify. This is outlined in `REQUEST_RESPONSE_ENCODING.MD` linked above the deck. In essence the body may includes padding to fit size requirements. The length then indicates which meaningful length (excluding padding).

Reply:
Unfortunately yes, though this is a fair point. In essence, the ABI that the wasm module interacts still uses proto. We should be able to remove the invocation part at least though.